test/test_js_client.py

```diff
@@ -0,0 +1,115 @@
+import textwrap
+import unittest
+
+from stone.backends.js_client import JavascriptClientBackend
+from stone.frontend.frontend import specs_to_ir
+from test.backend_test_util import _mock_emit
+
+MYPY = False
+if MYPY:
+    import typing  # noqa: F401 # pylint: disable=import-error,unused-import,useless-suppression
+
+
+class TestGeneratedJSClient(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        # type: (ApiNamespace) -> typing.Text
+        super(TestGeneratedJSClient, self).__init__(*args, **kwargs)
+
+    def test_route_versions(self):
+        # type: () -> None
+        text = textwrap.dedent("""\
```

Comment:
I'm thinking maybe we should consider constructing IR directly instead of from Stone using frontend. The unit tests for the backend should not have dependency on the frontend. This way if the tests fail we know there is something wrong on the backend.